cmapi/mcs_cluster_tool/cluster_app.py

```diff
@@ -32,9 +44,119 @@ def status():
+                'Use this with caution: on a working cluster '
+                'low values could cause data loss. '
+                'In interactive mode means delay time between promts.'
+            )
+        )
+    ] = 15,
+    force: Annotated[
+        bool,
+        typer.Option(
+            '--force/--no-force', '-f/-no-f',
+            help=(
+                'Force stopping a cluster.'
+                'WARNING: It could cause data corruption or/and data loss.'
+            ),
+            #TODO: hide from help till not investigated in decreased timeout
+            #      affect
+            hidden=True
+        )
+    ] = False
+):
     """Stop the Columnstore cluster."""
-    return ClusterHandler.shutdown(logger=logger)
+
+    start_time = str(datetime.now())
+    if interactive:
+        # TODO: for standalone cli tool need to change primary detection
+        #       method. Partially move logic below to ClusterController
+        nc = NodeConfig()
+        root = nc.get_current_config_root(
+            config_filename=DEFAULT_MCS_CONF_PATH
+        )
+        primary_node = root.find("./PrimaryNode").text
+        cfg_parser = get_config_parser(CMAPI_CONF_PATH)
+        api_key = get_current_key(cfg_parser)
+        version = get_version()
+
+        headers = {'x-api-key': api_key}
+        body = {'force': False, 'timeout': timeout}
+        url = f'https://{primary_node}:8640/cmapi/{version}/node/stop_dmlproc'
+        try:
+            resp = requests.put(
+                url, verify=False, headers=headers, json=body,
+                timeout=timeout+1
+            )
+            resp.raise_for_status()
+        except Exception as err:
+            raise CMAPIBasicError(
+                f'Error while stopping DMLProc on primary node.'
+            ) from err
+
+        force = True
+        while True:
+            time.sleep(timeout)
+            url = build_url(
+                base_url=primary_node, port=8640,
+                query_params={'process_name': 'DMLProc'},
+                path=f'cmapi/{version}/node/is_process_running',
+            )
+            try:
+                resp = requests.get(
+                    url, verify=False, headers=headers, timeout=timeout
+                )
+                resp.raise_for_status()
+            except Exception as err:
+                raise CMAPIBasicError(
+                    f'Error while getting mcs DMLProc status.'
+                ) from err
+
+            # check DMLPRoc state
+            # if ended, show message and break
+            dmlproc_running = resp.json()['running']
+            if not dmlproc_running:
+                logging.info(
+                    'DMLProc stopped gracefully. '
+                    'Continue stopping other processes.'
+                )
+                break
+            else:
+                force = typer.confirm(
+                    'DMLProc still working. '
```

Comment:
Idea
DMLProc is still running.
Do you want to force stop?
WARNING: Could cause data loss and or broken cluster

Reply:
Done.